agent/config/file_watcher.go

```diff
@@ -192,14 +219,17 @@ func (w *FileWatcher) isWatched(filename string) (*watchedFile, string, bool) {
 		// try to see if the watched path is the parent dir
 		newPath := filepath.Dir(path)
 		w.logger.Trace("get dir", "dir", newPath)
+		w.configFilesLock.RLock()
+		defer w.configFilesLock.RUnlock()
 		configFile, ok = w.configFiles[newPath]
```

Comment:
```suggestion
		w.configFilesLock.RLock()
		configFile, ok = w.configFiles[newPath]
		w.configFilesLock.RUnlock()
```

This also doesn't need to be deferred.

Consider a helper method:

```
func (w *FileWatcher) getConfigFile(path string) (*watchedFile, bool) {
	w.configFilesLock.RLock()
	defer w.configFilesLock.RUnlock()
	configFile, ok := w.configFiles[newPath]
	return configFile, ok
}
```